rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeperServer.java

```diff
@@ -82,9 +83,32 @@ public void run()
     return 0;
   }
 
-  public void startServer(Configuration conf, MetricRegistry metricsRegistry)
+  public BookKeeper startServer(final Configuration conf, MetricRegistry metricsRegistry)
   {
-    BookKeeper localBookKeeper;
+    setupServer(conf, metricsRegistry);
+    if (CacheConfig.isEmbeddedModeEnabled(conf)) {
+      new Thread(new Runnable()
+      {
+        @Override
+        public void run()
+        {
+          startThriftServer(conf, localBookKeeper);
+        }
+      }).start();
+    }
+    else {
+      startThriftServer(conf, localBookKeeper);
+    }
+
+    return localBookKeeper;
+  }
+
+  public void setupServer(Configuration conf, MetricRegistry metricsRegistry)
+  {
+    conf = new Configuration(conf);
+    CacheConfig.setCacheDataEnabled(conf, false);
```

Comment:
Shouldn't we disable Rubix only if it is in embeddedMode?

Reply:
This is disabling usage of cache inside the bookkeeperserver if somehow we end up with CachingFS object for FileSystem.